src/trajectory.jl

```diff
@@ -508,7 +513,7 @@ function update(
     τ::AbstractProposal,
     pc::Adaptation.AbstractPreconditioner
 ) 
-    metric = reconstruct(h.metric, M⁻¹=getM⁻¹(pc))
```

Comment:
Baiscally before only the new `M⁻¹` was passed to the new `metric` and `sqrtM⁻¹` or `cholM⁻¹` was not even re-computed @yebai.